internal/lock.go

```diff
@@ -149,11 +160,131 @@ func (l *Lock) Download(dir string, tags []string, notags []string, perm string)
 		return fmt.Errorf("nothing to download")
 	}
 	errorCh := make(chan error, total)
-	for _, r := range filteredResources {
+
+	byteCh := make(chan int64)
+	resourceSizes := make([]int64, 0, len(filteredResources))
+
+	if bar {
+		startTime := time.Now()
+
+		//Loop through resources, fetching their metadata and totalling up their sizes in bytes.
+		//This is not in a goroutine because a resource may finish downloading before its size has been calculated.
+		//For now, we'll first calculate sizes, then begin downloading.
+		fmt.Print(Color_Text("\rFetching file sizes...", "yellow"))
+
+		var totalBytes int64 = 0
+		for _, r := range filteredResources {
+			resource := r
+			httpClient := &http.Client{Timeout: 10 * time.Second}
+			resp, err := httpClient.Head(resource.Urls[0])
+			if err != nil {
+				//errorCh <- err	We don't want a failed fetch to crash the whole program -- it is not crucial to know the total download size.
+				//Instead, the print goroutine will check length of resourceSizes and notify the user by printing an error message in place of the byte counter.
+				break
```

Comment:
Since the size calculation is going to be wrong in case of errors, you probably want to not display the size at all in this case.